pilot/pkg/networking/core/v1alpha3/listener_waypoint.go

```diff
@@ -144,39 +139,21 @@ func (lb *ListenerBuilder) buildWaypointInboundTerminateConnect(svcs map[host.Na
 	}
 
 	// it's possible for us to hit this listener and target a Pod directly; route through the inbound-pod internal listener
-	// TODO: this shouldn't match on port; we should accept traffic to any port.
-	for _, wlx := range wls {
-		wl := wlx.WorkloadInfo
-		// TODO: fake proxy is really bad. Should have these take in Workload or similar
-		instances := lb.Discovery.GetProxyServiceInstances(&model.Proxy{
-			Type:            model.SidecarProxy,
-			IPAddresses:     []string{wl.PodIP},
-			ConfigNamespace: wl.Namespace,
-			Metadata: &model.NodeMetadata{
-				Namespace: wl.Namespace,
-				Labels:    wl.Labels,
-			},
-		})
-		// For each port, setup a route
-		for _, port := range getPorts(instances) {
-			clusterName := model.BuildSubsetKey(model.TrafficDirectionInboundPod, "internal", host.Name(wl.PodIP), port.Port)
-			vhost.Routes = append(vhost.Routes, &route.Route{
-				Match: &route.RouteMatch{
-					PathSpecifier: &route.RouteMatch_ConnectMatcher_{ConnectMatcher: &route.RouteMatch_ConnectMatcher{}},
-					Headers: []*route.HeaderMatcher{
-						istiomatcher.HeaderMatcher(":authority", fmt.Sprintf("%s:%d", wl.PodIP, port.Port)),
-					},
-				},
-				Action: &route.Route_Route{Route: &route.RouteAction{
-					UpgradeConfigs: []*route.RouteAction_UpgradeConfig{{
-						UpgradeType:   "CONNECT",
-						ConnectConfig: &route.RouteAction_UpgradeConfig_ConnectConfig{},
-					}},
-					ClusterSpecifier: &route.RouteAction_Cluster{Cluster: clusterName},
-				}},
-			})
-		}
-	}
+	vhost.Routes = append(vhost.Routes, &route.Route{
```

Comment:
I think we will probably want to limit this to only pods served by the waypoint, rather than all pods. otherwise I think we can use the waypoint to forward arbitrary requests to arbitrary destinations using the waypoint's identity (SSRF)

Reply:
Yeah, I think we only want to do this if the downstream and upstream identities match. In that case, I think it's valid to have a TLS bump.

Reply:
By upstream I mean: passthrough is allowed if waypoint and backend endpoint share an identity on the wire (we don't need to know which IPs have it). E.g. if a request is already authenticated at waypoint, then it's ok to forward to any endpoint with the same identity.

Reply:
Added verified peer SAN policy to upstream.